ci/test_plugins.rb

```diff
@@ -145,50 +142,95 @@ def install_gem(gem_file)
-               "logstash-input-redis", "logstash-input-stdin", "logstash-input-syslog", "logstash-input-udp",
-             ],
-    :codec => ["logstash-codec-avro", "logstash-codec-cef", "logstash-codec-es_bulk", "logstash-codec-json",
-               "logstash-codec-json_lines", "logstash-codec-line", "logstash-codec-multiline", "logstash-codec-plain",
-               "logstash-codec-rubydebug"],
-    :filter => ["logstash-filter-cidr", "logstash-filter-clone", "logstash-filter-csv", "logstash-filter-date", "logstash-filter-dissect",
-                "logstash-filter-dns", "logstash-filter-drop", "logstash-filter-elasticsearch", "logstash-filter-fingerprint",
-                "logstash-filter-geoip", "logstash-filter-grok", "logstash-filter-http", 
-                "logstash-filter-json", 
-                "logstash-filter-kv",
-                "logstash-filter-memcached", "logstash-filter-mutate", "logstash-filter-prune", "logstash-filter-ruby",
-                "logstash-filter-sleep", "logstash-filter-split", "logstash-filter-syslog_pri", "logstash-filter-translate",
-                "logstash-filter-truncate", "logstash-filter-urldecode", "logstash-filter-useragent", "logstash-filter-uuid",
-                "logstash-filter-xml",
-                ["logstash-filter-elastic_integration", "git@github.com:elastic/logstash-filter-elastic_integration.git"]
-              ],
-    :output => ["logstash-output-elasticsearch", "logstash-output-email", "logstash-output-file", "logstash-output-http",
-                "logstash-output-redis", "logstash-output-stdout", "logstash-output-tcp", "logstash-output-udp"],
-    :integration => ["logstash-integration-jdbc", "logstash-integration-kafka", "logstash-integration-rabbitmq",
-                     "logstash-integration-elastic_enterprise_search", "logstash-integration-aws"]
-  },
-  :tier2 => {
-    :input => [# "logstash-input-couchdb_changes", # Removed because of https://github.com/logstash-plugins/logstash-input-couchdb_changes/issues/51
-               "logstash-input-gelf", "logstash-input-graphite", "logstash-input-jms",
-                  "logstash-input-snmp", "logstash-input-sqs", "logstash-input-twitter"],
-    :codec => [#"logstash-codec-collectd", # Removed because of https://github.com/logstash-plugins/logstash-codec-collectd/issues/32
-               "logstash-codec-dots", "logstash-codec-fluent", "logstash-codec-graphite",
-                  "logstash-codec-msgpack", 
-                  "logstash-codec-netflow"
-                ],
-    :filter => ["logstash-filter-aggregate", "logstash-filter-de_dot", "logstash-filter-throttle"],
-    :output => ["logstash-output-csv", "logstash-output-graphite"]
-  },
-  :unsupported => {
-    :input => ["logstash-input-rss"]
-  }
-}
+# Models plugin's metadata, organization is optional, if nil then it consider logstash-plugins as default.
+PluginDefinition = Struct.new(:name, :support, :type, :organization)
+PLUGIN_DEFINITIONS = [
```

Comment:
I'd like to avoid having yet another list of plugins, but to avoid prolonging the merge of this PR I'd suggest creating a separate issue to reuse/expand the list in https://github.com/elastic/logstash/blob/main/rakelib/plugins-metadata.json

Reply:
I agree, using one source of truth and not hardcoded in a script is the best way, we already have such issue: #15190